ksp_plugin_test/pile_up_test.cpp

```diff
@@ -596,6 +597,50 @@ TEST_F(PileUpTest, Serialization) {
   EXPECT_THAT(message, EqualsProto(second_message));
 }
 
+TEST_F(PileUpTest, SerializationCompatibility) {
+  MockEphemeris<Barycentric> ephemeris;
+  p1_.increment_intrinsic_force(
+      Vector<Force, Barycentric>({1 * Newton, 2 * Newton, 3 * Newton}));
+  p2_.increment_intrinsic_force(
+      Vector<Force, Barycentric>({11 * Newton, 21 * Newton, 31 * Newton}));
+  TestablePileUp pile_up({&p1_, &p2_},
+                         astronomy::J2000,
+                         DefaultProlongationParameters(),
+                         DefaultHistoryParameters(),
+                         &ephemeris);
+
+  serialization::PileUp message;
+  pile_up.WriteToMessage(&message);
+
+  // Clear the children to simulate pre-Cesàro serialization.
+  message.mutable_history()->clear_children();
+
+  auto const part_id_to_part = [this](PartId const part_id) {
+    if (part_id == part_id1_) {
+      return &p1_;
+    }
+    if (part_id == part_id2_) {
+      return &p2_;
+    }
+    LOG(FATAL) << "Unexpected part id " << part_id;
+    base::noreturn();
+  };
+  auto p = PileUp::ReadFromMessage(message, part_id_to_part, &ephemeris);
+
+  EXPECT_CALL(ephemeris, FlowWithAdaptiveStep(_, _, _, _, _, _))
```

Comment:
Are we actually *expecting* this call? it seems we should use `ON_CALL` instead of pretty much all instances of `EXPECT_CALL` in this file. That is probably a job for a later pull request though.